### ISSUE TITLE
Online Co-op: ERROR: No case for PlayerStruct HeaderByte 230

### ISSUE THREAD

**Issue by ChrisNZL** [ChrisNZL/Tallowmere2#848]
Auto report from two players in same game. v0.2.3.

Server: Vultr-Toronto-01
Game ID: 5XW23

Feedback IDs: 20210130-RRB3Y, 20210130-KPP2B

Something after all players had died; maybe an issue trying to start a new game afterwards.

No errors on server log.

Client log:

```
6:52:40, Frame 312057, ERROR »  ERROR: No case for PlayerStruct HeaderByte 230
>>>>> CRITICAL ERROR >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>


GameStates: InitialisingAsNetworkHost, RompingThroughDungeon, TransitioningFromGameOverToHighScores, ViewingHighScoreCanvas
OnlineGameInfo: Device is online.
Server: Vultr-Toronto-01
GameID: 5XW23
GameSetupMode: NetworkCoop
Players: 2
DungeonRoomTransitionerState: Null
Room: 19 / 19
RoomModifiers: AxesOnly (大斧派对), Spiky (满屋尖刺), MoreGoldOnDeath (财富)
YouHaveDiedState: Visible
SystemPlayer InputDevice: 键盘
HumanPlayer 1 InputDevice: 键盘
```

**Comment by ChrisNZL** [comment]
Further feedback IDs:
- 20210215-6WVTF
- 20210215-8RYW9
- 20210215-Z9LPY
- 20210215-TXZZN

Server: Lightning-Beijing-01
Game ID: 7ZDPJ

**Comment by ChrisNZL** [comment]
Addressed, or at least handled better, in 0.2.5.

Added some debugging trace calls within the PlayerStruct constructor. Will be an ongoing investigation.

Suspect it might be an issue with players disconnecting or leaving while the game is restarting online.

May need to refactor the online dungeon-restart flow if it's an ongoing issue.

Things to test further:
- [ ] Player name lengths
- [ ] Test having player(s) disconnect at ~same time as host chooses to restart.
- [ ] Test captain spamming the reconnect button ASAP after party dies.
